cmake/FindZSTD.cmake

```diff
@@ -0,0 +1,48 @@
+#[=======================================================================[.rst:
+FindZSTD
+-----------
```

Comment:
Please excuse me for being nit picky…

Where is this file coming from? What's the copyright?

Or do I miss something?

Reply:
@ChristianTackeGSI,

the file does not come from somewhere but was inspired by FindLibLZMA.cmake, so I took the same documentation block used by official  modules. 
You are right, I should add our copyright.

Reply:
Fixed with last commit. 
Also add extracting version info in case zstd binary is not available.